src/amp-story-player/amp-story-player-impl.js

```diff
@@ -730,22 +785,84 @@ export class AmpStoryPlayer {
 
   /**
    * Indicates the player changed story.
+   * @param {!Object} data
    * @private
    */
-  signalNavigation_() {
-    const index = this.currentIdx_;
-    const remaining = this.stories_.length - this.currentIdx_ - 1;
+  signalNavigation_(data) {
     const event = createCustomEvent(
       this.win_,
       'navigation',
-      dict({
-        'index': index,
-        'remaining': remaining,
-      })
+      /** @type {!JsonObject} */ (data)
     );
     this.element_.dispatchEvent(event);
   }
 
+  /**
+   * Triggers when swithing from one story to another.
+   * @private
+   */
+  onNavigation_() {
+    const index = this.currentIdx_;
+    const remaining = this.stories_.length - this.currentIdx_ - 1;
+    const navigation = {
+      'index': index,
+      'remaining': remaining,
+    };
+
+    this.signalNavigation_(navigation);
+    if (this.playerConfig_ && this.playerConfig_.behavior) {
+      this.maybeFetchMoreStories_(remaining);
+    }
+  }
+
+  /**
+   * Fetches more stories if appropiate.
+   * @param {number} remaining Number of stories remaining in the player.
+   */
+  maybeFetchMoreStories_(remaining) {
+    if (
+      remaining <= FETCH_STORIES_THRESHOLD &&
+      this.shouldFetchMoreStories_()
+    ) {
+      this.fetchStories_()
+        .then((stories) => {
+          if (!stories) {
+            return;
+          }
+          this.fetchStoriesOffset_ += stories.length;
```

Comment:
I think `fetchStoriesOffset_` always has to be `this.stories_.length`. If you have 10 initial stories, the initial `offset` is `10`

Reply:
Not super familiar with how `offset` is usually used, so feel free to correct. 

When the player has 1 story and the endpoint contains 10. The publisher would want the offset to start at 0 from the endpoint, right? Because this wouldn't be the case if we used `this.stories_.length`.

Reply:
Discussed offline and we'll take into account the stories that were already in the player for the offset property.